numalogic/udfs/preprocess.py

```diff
@@ -95,17 +95,16 @@ def exec(self, keys: list[str], datum: Datum) -> Messages:
         except (orjson.JSONDecodeError, KeyError):  # catch json decode error only
             _LOGGER.exception("Error while decoding input json")
             return Messages(Message.to_drop())
-        raw_df, timestamps = get_df(
-            data_payload=data_payload, stream_conf=self.get_conf(data_payload["config_id"])
-        )
+
+        _stream_conf = self.get_stream_conf(data_payload["config_id"])
+        _conf = _stream_conf.ml_pipelines[data_payload["pipeline_id"]]
+        raw_df, timestamps = get_df(data_payload=data_payload, stream_conf=_stream_conf)
 
         _metric_label_values = (self._vtx, ":".join(keys), data_payload["config_id"])
```

Comment:
I think for metrics we need to have pipeline_id also an label. Can you please add a TODO and open an issue and assign it to me.